src/mpid/ch4/netmod/ofi/ofi_probe.h

```diff
@@ -64,9 +64,12 @@ static inline int MPIDI_OFI_do_iprobe(int source,
     }
 
     MPIDI_OFI_CALL(ofi_err, trecvmsg);
-    MPIDI_OFI_PROGRESS_WHILE(MPIDI_OFI_REQUEST(rreq, util_id) == MPIDI_OFI_PEEK_START);
+    /* Block until the peek request is processed by netmod */
+    MPIDI_OFI_PROGRESS_WHILE(MPL_atomic_acquire_load_int(&(MPIDI_OFI_REQUEST(rreq, util_id))) ==
+                             MPIDI_OFI_PEEK_START);
 
-    switch (MPIDI_OFI_REQUEST(rreq, util_id)) {
+    /* Ordering constraint for util_id is unnecessary after the thread unblocks */
+    switch (MPL_atomic_relaxed_load_int(&(MPIDI_OFI_REQUEST(rreq, util_id)))) {
```

Comment:
Ditto here with `MPL_atomic_load_int`.

Reply:
Oh, nevermind this is actually intended to be relaxed.

Reply:
Yea, this was supposed to be relaxed.